src/blob/index/core.rs

```diff
@@ -302,21 +302,30 @@ where
         }
     }
 
-    async fn get_any(&self, key: &K) -> Result<Option<RecordHeader>> {
+    async fn get_any(&self, key: &K) -> Result<ReadResult<RecordHeader>> {
```

Comment:
`get_all` above should also be updated to return `Result<ReadResult<Vec<RecordHeader>>>`